app/services/usps_in_person_proofing/proofer.rb

```diff
@@ -112,12 +107,14 @@ def request_enrollment_code(unique_id)
     # @return [String] the token
     def retrieve_token!
       body = request_token
-      @token_expires_at = Time.zone.now + body['expires_in']
-      @token = "#{body['token_type']} #{body['access_token']}"
+      @@token_expires_at = Time.zone.now + body['expires_in']
+      @@token = "#{body['token_type']} #{body['access_token']}"
     end
 
     def token_valid?
-      @token.present? && @token_expires_at.present? && @token_expires_at.future?
+      @@token.present? &&
+        @@token_expires_at.present? &&
+        (@@token_expires_at - 1.second).future?
```

Comment:
Aiming to avoid the edge case that the token is valid when we check it but invalid by the time the request is processed

Reply:
I was kinda wondering about this, like if it's something where we could build in token retrieval + retry if the token was invalid for whatever reason (expired due to race condition or any other factor on USPS' end). Technically we could even go as far as to get rid of any consideration of expiration altogether, and just wait for the token to start failing on us before we retrieve the next one.

Reply:
Yeah. I think that's out of scope of this ticket, just want to get this minor fix in before we do any load testing. But happy to create another ticket

Maybe something like replacing all instances of `faraday.<verb>` with `<verb>` and having helpers like this?

```
def self.<verb>(**args) // syntax may be off, need to google ruby argument grouping syntax
  begin
    faraday.<verb>(**args)
  rescue Faraday::ClientError => err
    if err.status == UNAUTHORIZED_CODE && err.message == TOKEN_EXPIRED_MESSAGE
      retrieve_token!
      faraday.<verb>(**args)
    else
      raise
    end
end
```

Reply:
Okay, reverted in d1eb342f9. I'll create a ticket for what we talked about

Reply:
Created [this ticket](https://cm-jira.usa.gov/browse/LG-7836) for the work above